### PR TITLE
Fix inconsistent behaviour of --dump_python

### PR DESCRIPTION
**Pull request by pkalbhor**
#### PR description:
Bug fix for `cmsDriver.py` which when used with `--dump_python` was giving inconsistent/incorrect output. Issue was raised few times already [Issue#37976](https://github.com/cms-sw/cmssw/issues/37976), [Issue#38686](https://github.com/cms-sw/cmssw/issues/38686)

P.S. My first PR to cmssw :)

### CODE REVIEW

**Comment by qliphy** [comment]
please test

**Comment by makortel** [comment]
> I am not sure that this is the correct assignment for this PR

It is not :)

Anyway, I think it is not worse than the current implementation (and certainly works around #37976). Using `edmConfigDump` to read and write the same file smells a bit, but after looking the implementation it is not worse than what `cmsDriver.py` does already, i.e. keep the read file handle open while opening the file again for writing (while truncating it).

I think the `cmsDriver.py` should check the return value of `edmConfigDump`, and if non-zero, terminate the program immediately with the same return value.

**Comment by rappoccio** [comment]
@makortel can you confirm with the usual "+1" if you're okay with this (as discussed in ORP meeting)?

**Comment by makortel** [comment]
I'd like to see

> I think the `cmsDriver.py` should check the return value of `edmConfigDump`, and if non-zero, terminate the program immediately with the same return value.

addressed first.

**Comment by srimanob** [comment]
@pkalbhor 
Could you please provide the fix following the suggestion of @makortel [here](https://github.com/cms-sw/cmssw/pull/39127#issuecomment-1230498979)
Thanks.

**Comment by pkalbhor** [comment]
Hi @makortel I hope you are expecting these changes only
replaced
```python
        if options.dump_python:
            os.system('edmConfigDump -o {f} {f}'.format(f=options.
```
with 
```python
        if options.dump_python:
            status_code = os.system('edmConfigDump -o {f} {f}'.format(f=options.python_filename))
            if status_code: exit()
```

File: Configuration/Applications/scripts/cmsDriver.py [hunk 1]
```diff
@@ -33,13 +33,8 @@ def run():
...
         if options.dump_python:
-            result = {}
-            exec(open(options.python_filename).read(), result)
-            process = result["process"]
-            expanded = process.dumpPython()
-            expandedFile = open(options.python_filename,"w")
-            expandedFile.write(expanded)
-            expandedFile.close()
+            status_code = os.system('edmConfigDump -o {f} {f}'.format(f=options.python_filename))
+            if status_code: exit()
```
Comment (makortel) [inline Configuration/Applications/scripts/cmsDriver.py:37]: I'd suggest
```suggestion
            if status_code: sys.exit(status_code)
```

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please test

To clean up the comparisons

**Comment by makortel** [comment]
+core

(with the disclaimer that this package is not under `core` responsibility)

**Comment by perrotta** [comment]
+1

**Pull request merged**